db/components.go

```diff
@@ -0,0 +1,115 @@
+package db
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/flanksource/commons/logger"
+	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/db/types"
+	"github.com/flanksource/incident-commander/utils"
+	"github.com/google/uuid"
+	"gorm.io/gorm/clause"
+)
+
+func GetTeamsWithComponentSelector() map[uuid.UUID][]api.ComponentSelector {
+	var teams []api.Team
+	var teamComponentMap = make(map[uuid.UUID][]api.ComponentSelector)
+	err := Gorm.Table("teams").Where("spec::jsonb ? 'components';").Find(&teams).Error
+	if err != nil {
+		logger.Errorf("error fetching the teams with componenets: %v", err)
+	}
+	for _, team := range teams {
+		teamSpec := &api.TeamSpec{}
+		teamSpecJson, err := team.Spec.MarshalJSON()
+		if err != nil {
+			logger.Errorf("error marshalling team spec for team: %v", team.ID)
+			continue
+		}
+		if err := json.Unmarshal(teamSpecJson, teamSpec); err != nil {
+			logger.Errorf("error unmarshalling the teamSpec for team: %v", team.ID)
+		}
+		teamComponentMap[team.ID] = teamSpec.Components
+	}
+	return teamComponentMap
+}
+
+func GetComponentsWithSelectors(componentSelectors []api.ComponentSelector) map[string][]uuid.UUID {
+	var selectedComponents = make(map[string][]uuid.UUID)
+	for _, compSelector := range componentSelectors {
+		selectedComponents[utils.GetHash(compSelector)] = getComponentsWithSelector(compSelector)
+	}
+
+	return selectedComponents
+}
+
+func getComponentsWithSelector(selector api.ComponentSelector) []uuid.UUID {
+	sql := "select ID from components where deleted_at is null "
```

Comment:
use `query` variable since `database/sql` is a stdlib package and we are using it in this project

Reply:
not applicable anymore